rero_ils/modules/loans/logs/api.py

```diff
@@ -0,0 +1,209 @@
+    def _get_document_data(cls, document_pid):
+        """Get document record data.
+
+        :param str document_pid: Document PID
+        :returns: Document formatted data
+        :rtype: dict
+        """
+        document = Document.get_record_by_pid(document_pid)
+        document = document.dumps()
+        return {
+            'title':
+            document['title'][0]['_text'],
+            'type':
+            document['type'][0].get('subtype',
+                                    document['type'][0]['main_type'])
+        }
+
+    @classmethod
+    def _get_holding_data(cls, holding_pid):
+        """Get holding record data.
+
+        :param str holding_pid: Holding PID
+        :returns: Holding formatted data
+        :rtype: dict
+        """
+        holding = Holding.get_record_by_pid(holding_pid)
+
+        return {
+            'pid':
+            holding['pid'],
+            'location_name':
+            cls._get_location_name(
+                extracted_data_from_ref(holding['location']['$ref']))
+        }
+
+    @classmethod
+    def _get_location_name(cls, location_pid):
+        """Get location name for a location PID.
+
+        :param str location_pid: Location PID
+        :returns: Location name
+        :rtype: str
+        """
+        location = Location.get_record_by_pid(location_pid)
+        return location['name']
+
+    @classmethod
+    def _get_patron_data(cls, patron_pid):
+        """Get patron record data.
+
+        :param str patron_pid: Patron PID
+        :returns: Patron formatted data
+        :rtype: dict
+        """
+        patron = Patron.get_record_by_pid(patron_pid)
+
+        patron_type = None
+
+        if patron.get('patron'):
+            patron_type = PatronType.get_record_by_pid(
+                extracted_data_from_ref(patron['patron']['type']['$ref']))
+
+        return {
+            'name': patron.formatted_name,
+            'type': patron_type['name'] if patron_type else None,
+            'birth_date': str(patron.user.profile.birth_date),
+            'postal_code': patron.user.profile.postal_code,
+            'gender': patron.user.profile.gender or 'other'
+        }
+
+    @classmethod
+    def get_logs_by_record_pid(cls, pid):
+        """Get all logs for a given record PID.
+
+        :param str pid: record PID.
+        :returns: List of logs.
+        :rtype: list
+        """
+        return list(
+            RecordsSearch(index=cls.index_name).filter(
+                'bool', must={
+                    'exists': {
+                        'field': 'loan'
+                    }
+                }).filter('term', record__pid=pid).scan())
+
+    @classmethod
+    def anonymize_logs(cls, loan_pid):
+        """Anonymize all logs corresponding to the given loan.
+
+        :param loan_pid: Loan PID.
+        """
+        for log in cls.get_logs_by_record_pid(loan_pid):
+            record = log.to_dict()
+            record['record']['patron_pid'] = uuid.uuid1()
```

Comment:
I would be great to have the same id for the same `patron_pid`: a one-way function like an hash.